Exercises.Delegate/Exercises.Delegate/Point.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System;
+using System.Collections;
+
+namespace Exercises.Delegate
+{
+    public class Point : IComparable
+    {
+
+        public enum PointRepresentation { Polar, Rectangular }
+
+        private double r, a;
+
+        public Point(double x, double y)
+        {
+            r = RadiusGivenXy(x, y);
+            a = AngleGivenXy(x, y);
+        }
+
+        public Point(double par1, double par2, PointRepresentation pr)
+        {
+            if (pr == PointRepresentation.Polar)
+            {
+                r = par1; a = par2;
+            }
+            else
+            {
+                r = RadiusGivenXy(par1, par2);
+                a = AngleGivenXy(par1, par2);
+            }
+        }
+
+        public double X
+        {
+            get { return XGivenRadiusAngle(r, a); }
+        }
+
+        public double Y
+        {
+            get { return YGivenRadiusAngle(r, a); }
+        }
+
+
+        public double Radius
+        {
+            get { return r; }
+        }
+
+        public double Angle
+        {
+            get { return a; }
+        }
+
+        public void Move(double dx, double dy)
+        {
+            double x, y;
+            x = XGivenRadiusAngle(r, a); y = YGivenRadiusAngle(r, a);
+            r = RadiusGivenXy(x + dx, y + dy);
+            a = AngleGivenXy(x + dx, y + dy);
+        }
+
+        public void Rotate(double angle)
+        {
+            a += angle;
+        }
+
+        public override string ToString()
+        {
+            return "(" + XGivenRadiusAngle(r, a) + "," + YGivenRadiusAngle(r, a) + ")";
+        }
+
+
+        private static double RadiusGivenXy(double x, double y)
```

Comment:
All methods below need better naming. Method returns something, so pattern for this is to start with `Get...`.
`Xy` can be replaced with `Coordinates`
There should not be (usually) two subsequent uppercase letters: `XGiven...` -> `Xgiven...`

Proposal:
`GetRadius`, `GetRadiusFrom`, `CalculateRadiusBy`, `GetRadiusFromCoordinates`, etc.